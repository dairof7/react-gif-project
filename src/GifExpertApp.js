import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    //puedo usar las categorias de esta forma si nunca se van a modificar
    //de lo contraria es mejor usar hooks
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState([''])
    console.log('la app')
    return (
        <>
            <h2>Gif Search</h2>
            {/* enviamos la funcion setcategories al argumento con el mismo nombre */}
            <AddCategory setCategories={ setCategories } />
            <hr/>
            <ol>
                {
                    categories.map( category => (
                    //react necesita que cada lista se asocie a un key
                    //en este caso la key va a ser cada uno de los elementos
                    //el key normalmente es el id proveniente de una base de datos
                    //return <li key= { category }> { category }</li> 
                    //ahora reemplazamos por otro componente
                    <GifGrid 
                        key={category}
                        category= {category} />
                    ))
                }
            </ol>
        </>
    )
}
