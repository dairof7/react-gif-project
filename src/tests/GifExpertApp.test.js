const { shallow } = require("enzyme")
import React from 'react'
import { GifExpertApp } from '../GifExpertApp'

describe('test <GifExpertApp />', () => {
    test('should show correctly', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('should show a list of categories', () => {
        
        const categories = ['hola', 'chavo']

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />)

        expect( wrapper ).toMatchSnapshot()
        expect ( wrapper.find('GifGrid').length ).toBe( categories.length )

    })
    
})
