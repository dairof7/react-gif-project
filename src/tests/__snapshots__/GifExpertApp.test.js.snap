// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifExpertApp /> should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="hola"
      key="hola"
    />
    <GifGrid
      category="chavo"
      key="chavo"
    />
  </ol>
</Fragment>
`;

exports[`test <GifExpertApp /> should show correctly 1`] = `
<Fragment>
  <h2>
    Gif Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
